File: doc/examples/obconversion_readstring.cpp
#include <openbabel/obconversion.h>
#include <openbabel/mol.h>

using namespace OpenBabel;

int main()
{
  // Create an OBConversion object.
  OBConversion conv;
  // Set the input format.
  if (!conv.SetInFormat("smi")) {
    // Handle error.
    return 1;
  }

  // Create the OBMol object.
  OBMol mol;

  // Read the smiles string.
  if (conv.ReadString(&mol, "CCCC")) {
    // Handle error.
    return 1;
  }

  // ...Use OBMol object...
}

File: doc/examples/obconversion_readstring.py
from __future__ import print_function
from openbabel import openbabel as ob

# Initialize the OBConversion object
conv = ob.OBConversion()
if not conv.SetInFormat('smi'):
  print('could not find smiles format')

# Read the smiles string
mol = ob.OBMol()
if not conv.ReadString(mol, 'CCCC'):
  print('could not read the smiles string')

# ... Use OBMol object ...
